general/device.cpp

```diff
@@ -152,29 +157,66 @@ void Device::Print(std::ostream &out)
       out << "libCEED backend: " << ceed_backend << '\n';
    }
 #endif
+   out << std::endl;
+   out << "Memory configuration: "
+       << MemoryTypeName[static_cast<int>(host_mem_type)];
+   if (Device::Allows(Backend::DEVICE_MASK) ||
+       Device::Allows(Backend::DEBUG))
+   {
+      out << ", " << MemoryTypeName[static_cast<int>(device_mem_type)];
+   }
+   out << std::endl;
 }
 
 void Device::UpdateMemoryTypeAndClass()
 {
-   if (Device::Allows(Backend::DEVICE_MASK))
+#ifdef MFEM_USE_UMPIRE
+   const bool umpire = true;
+#else
+   const bool umpire = false;
+#endif
+   const bool debug = Device::Allows(Backend::DEBUG);
+   const bool device = Device::Allows(Backend::DEVICE_MASK);
+
+   // Enable the device memory type
+   if (device)
    {
-      mem_type = MemoryType::CUDA;
-      mem_class = MemoryClass::CUDA;
+      device_mem_type = MemoryType::DEVICE;
+      device_mem_class = MemoryClass::DEVICE;
    }
-   else
+
+   // If MFEM has been compiled with Umpire support, use it as the default
+   if (umpire)
+   {
+      host_mem_type = MemoryType::HOST_UMPIRE;
+      if (device) { device_mem_type = MemoryType::DEVICE_UMPIRE; }
+   }
+
+   // Enable the UVM shortcut when requested
+   if (getenv("UVM") && device)
```

Comment:
`UVM` is too generic to use here. I think `MFEM_UVM` will be better.

Reply:
I switched to `MFEM_USE_UVM`.

I wanted to keep the functionality to test the code with `UVM`.

This was previously done by  some pseudo `uvm` device given on the `device` command line option, but I reverted this in order not to mix memory backends selection with the device one.

Do we want to add such memory backend options?

Reply:
We can add memory backend options later.

Reply:
Added the `uvm` option for the `cuda` device.